pkg/webui/components/checkbox/group/index.js

```diff
@@ -0,0 +1,335 @@
+            .map(function (field) {
+              const fieldName = formatFieldName(name, field.props.name)
+              const valueProp = form ? {} : { value: selected.includes(fieldName) }
+              return React.cloneElement(field, {
+                ...field.props,
+                ...valueProp,
+                onChange: onSelectedChange,
+                type: 'checkbox',
+                form,
+              })
+            })}
+        </List>
+      </div>
+    )
+  }
+
+  get transfer () {
+    const {
+      transfer,
+      form,
+      children,
+      name,
+    } = this.props
+
+    if (!transfer) {
+      return null
+    }
+
+    const { selected } = this.state
+    const onTransferSelectedChange = this.onTransferSelectedChange
+
+    const fields = React.Children.toArray(children)
+    const transferFields = selected.map(function (selectedName) {
+      const field = fields.find(f => formatFieldName(name, f.props.name) === selectedName)
+      return React.cloneElement(field, {
+        ...field.props,
+        key: field.props.name,
+        type: 'checkbox',
+        onChange: onTransferSelectedChange,
+        value: true,
+        form,
+      })
+    })
+
+    return (
+      <div className={style.bodyContentSelected}>
+        <List
+          className={style.bodyContent}
+        >
+          {transferFields}
+        </List>
+      </div>
+    )
+  }
+
+  render () {
+    const {
+      className,
+      name,
+      title,
+      errors,
+      search,
+      transfer,
+    } = this.props
+
+    const bodyClassNames = classnames(style.body, {
+      [style.bodySearch]: search,
+      [style.bodyWithTransfer]: transfer,
+    })
+
+    return (
+      <FieldGroup
+        name={name}
+        title={title}
+        errors={errors}
+      >
+        <div className={classnames(className, style.checkboxGroup)}>
+          {this.header}
+          <div className={bodyClassNames}>
+            {this.all}
+            {this.transfer}
+          </div>
+        </div>
+      </FieldGroup>
+    )
+  }
+}
+
+const List = function (props) {
+  const {
+    className,
+    children,
+  } = props
+
+  const fields = React.Children.map(children, function (Child) {
+    return (
+      <li
+        className={style.bodyContentItem}
+        key={name}
+      >
+        {React.cloneElement(Child, { ...Child.props })}
+      </li>
+    )
+  })
+
+  return (
+    <ul className={className}>
+      {fields}
+    </ul>
+  )
+}
+
+CheckboxGroup.propTypes = {
+  form: PropTypes.bool,
+  search: PropTypes.bool,
+  transfer: PropTypes.bool,
+  onSearch: PropTypes.func,
+  searchDebounce: PropTypes.oneOfType([ PropTypes.number, PropTypes.bool ]),
+  searchPlaceholder: PropTypes.message,
+  name: PropTypes.string.isRequired,
+  selectAllTitle: PropTypes.message,
+  selectAllName: PropTypes.string,
+}
+
+CheckboxGroup.defaultProps = {
+  form: true,
+  search: false,
+  transfer: false,
+  onSearch: () => true,
```

Comment:
Shouldn't the search function you defined in the story be the default function rather?